torbot/modules/collect_data.py

```diff
@@ -58,16 +58,19 @@ def collect_data(user_url: str):
         writer = SafeDictWriter(outcsv, fieldnames=fieldnames)
         bar = Bar('Processing...', max=len(links))
         for link in links:
-            resp = requests.get(link)
-            soup = BeautifulSoup(resp.text, 'html.parser')
-            meta_tags = parse_meta_tags(soup)
-            entry = {
-                "ID": uuid.uuid4(),
-                "Title": soup.title.string,
-                "Metadata": meta_tags,
-                "Content": soup.find('body')
-            }
-            writer.writerow(entry)
+            try:
+                resp = requests.get(link)
+                soup = BeautifulSoup(resp.text, 'html.parser')
+                meta_tags = parse_meta_tags(soup)
+                entry = {
+                    "ID": uuid.uuid4(),
+                    "Title": soup.title.string if soup.title else "",
+                    "Metadata": meta_tags,
+                    "Content": soup.find('body')
+                }
+                writer.writerow(entry)
+            except requests.exceptions.RequestException:
```

Comment:
Could you grab the error and print it out here as well? You could use a debug if you don't want to clutter the console.

Reply:
Thanks for the review! ok, I'll fix it to use debug and print out exception!

Reply:
I fixed it ! https://github.com/DedSecInside/TorBot/pull/271/commits/b9a6f304f5fe04cd69d197c7590899f57d3a6095